test/unit/test_basicconfig.py

```diff
@@ -26,10 +27,10 @@ def test_invalid_expid_path():
                    BasicConfig.expid_log_dir,
                    BasicConfig.expid_aslog_dir]
 root_dirs = [
-    lambda root_path, exp_id: os.path.join(root_path, exp_id),
-    lambda root_path, exp_id: os.path.join(root_path, exp_id, "tmp"),
-    lambda root_path, exp_id: os.path.join(root_path, exp_id, "tmp", f"LOG_{exp_id}"),
-    lambda root_path, exp_id: os.path.join(root_path, exp_id, "tmp", "ASLOGS")
+    lambda root_path, exp_id: Path(root_path).joinpath(exp_id),
+    lambda root_path, exp_id: Path(root_path).joinpath(exp_id, "tmp"),
+    lambda root_path, exp_id: Path(root_path).joinpath(exp_id, "tmp", f"LOG_{exp_id}"),
+    lambda root_path, exp_id: Path(root_path).joinpath(exp_id, "tmp", "ASLOGS")
```

Comment:
These calls also don't need `joinpath`, and just the constructor should suffice, I think (not tested)